lib/cn/csched_sp3.c

```diff
@@ -978,11 +978,20 @@ sp3_dirty_node_locked(struct sp3 *sp, struct cn_tree_node *tn)
                 const uint64_t weight = ((uint64_t)sp->thresh.lcomp_split_pct << 32) | keys_uniq;
 
                 sp3_node_insert(sp, spn, wtype_split, weight);
-                ev_debug(1);
+                splitting = true;
             } else {
                 sp3_node_remove(sp, spn, wtype_split);
             }
         }
+
+        /* Node splits are rare, but when a node is ready to split it needs
+         * be done as soon as possible.  Hence splits preempt all other work.
+         */
+        if (splitting) {
+            sp3_node_remove(sp, spn, wtype_length);
+            sp3_node_remove(sp, spn, wtype_garbage);
+            sp3_node_remove(sp, spn, wtype_scatter);
```

Comment:
Does anything prevent these nodes from being put back on these work lists?

Reply:
They won't be re-considered until the node changes again (say after a splill or split).  But after a spill if split is still on the table then they will be removed again.   Would be better not to put them on in that case, but that requires a much bigger change.  Since split is rare I figured it's ok to just undo the work in this case..